programs/bpf_loader/src/lib.rs

```diff
@@ -1403,6 +1417,30 @@ fn process_loader_upgradeable_instruction(
                 .try_borrow_instruction_account(transaction_context, PROGRAM_DATA_ACCOUNT_INDEX)?;
             programdata_account.set_data_length(new_len)?;
 
+            let programdata_data_offset = UpgradeableLoaderState::size_of_programdata_metadata();
+
+            deploy_program!(
+                invoke_context,
+                program_key,
+                &program_id,
+                UpgradeableLoaderState::size_of_program().saturating_add(new_len),
```

Comment:
I think `new_len` should already be the full account size, why is `size_of_program` added here as well?

Reply:
I think you are right, this is a bug, probably copy pasta.
A fix should be backported before the feature is activated.

@pgarg66 Thoughts?